sky/engine/platform/fonts/apple/FontCacheIOS.mm

```diff
@@ -0,0 +1,178 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <Foundation/Foundation.h>
+#import <CoreFoundation/CoreFoundation.h>
+#import <CoreText/CoreText.h>
+#import <UIKit/UIKit.h>
+
+#include "sky/engine/platform/fonts/FontCache.h"
+#include "sky/engine/platform/fonts/FontCache.h"
+#include "base/mac/scoped_nsobject.h"
+#include "base/lazy_instance.h"
+#include "base/macros.h"
+#include "base/logging.h"
+
+namespace blink {
+
+template <class T>
+class CFRef {
+ public:
+  CFRef() : _ref(nullptr) {}
+
+  CFRef(T t) : _ref(t) {}
+
+  CFRef(const CFRef&) = delete;
+
+  void operator=(const CFRef&) = delete;
+
+  CFRef(CFRef&& o) : _ref(o._ref) { o._ref = nullptr; }
+
+  void operator=(CFRef&& o) {
+    _ref = o._ref;
+    o._ref = nullptr;
+  }
+
+  operator bool() const { return _ref != nullptr; }
+
+  T get() const { return _ref; }
+
+  ~CFRef() {
+    if (_ref != nullptr) {
+      CFRelease(reinterpret_cast<CFTypeRef>(_ref));
+    }
+  }
+
+ private:
+  T _ref;
+};
+
+class FontFallbackSelector {
+ public:
+  FontFallbackSelector() {
+    UIFontDescriptor* ui_desc =
+        [UIFont systemFontOfSize:[UIFont systemFontSize]].fontDescriptor;
+
+    CFRef<CTFontDescriptorRef> desc(CTFontDescriptorCreateWithNameAndSize(
+        reinterpret_cast<CFStringRef>(ui_desc.postscriptName),
+        ui_desc.pointSize));
+
+    if (!desc) {
+      return;
+    }
+
+    _prototype = CFRef<CTFontRef>{
+        CTFontCreateWithFontDescriptor(desc.get(), ui_desc.pointSize, nullptr)};
+  }
+
+  CFRef<CTFontDescriptorRef> fallbackFont(unichar glyph) {
+    if (!_prototype) {
+      return {};
+    }
+
+    CFRef<CFStringRef> unicode_string(CFStringCreateWithFormat(
+        kCFAllocatorDefault, nullptr, CFSTR("%C"), glyph));
+
+    if (!unicode_string) {
+      return {};
+    }
+
+    CFRef<CTFontRef> font(CTFontCreateForString(
+        _prototype.get(), unicode_string.get(),
+        CFRangeMake(0, CFStringGetLength(unicode_string.get()))));
+
+    if (!font) {
+      return {};
+    }
+
+    return CTFontCopyFontDescriptor(font.get());
+  }
+
+ private:
+  CFRef<CTFontRef> _prototype;
+
+  DISALLOW_COPY_AND_ASSIGN(FontFallbackSelector);
+};
+
+static base::LazyInstance<FontFallbackSelector> g_fallback_selector =
+    LAZY_INSTANCE_INITIALIZER;
+
+void FontCache::getFontForCharacter(
+    UChar32 c,
+    const char*,
```

Comment:
That is supposed to be the locale. But we are never passed one by the caller (it is always empty). Since it is not used in this function, it makes the linter not yell at you for declaring an unused argument.